src/cloudflare/internal/vectorize-api.ts

```diff
@@ -114,6 +114,45 @@ class VectorizeIndexImpl implements Vectorize {
     }
   }
 
+  public async queryById(
+    vectorId: string,
+    options?: VectorizeQueryOptions
+  ): Promise<VectorizeMatches> {
+    if (this.indexVersion === 'v1') {
+      throw new Error(`QueryById operation is not supported for v1 indexes.`);
+    } else {
+      if (options?.returnMetadata) {
```

Comment:
Can we factor out this duplicate code between query() and queryById()? Something like a queryImpl() that accepts an object that contains either a `vector` or `vectorId` field, then spreads the object into the body.

Reply:
Great suggestion! Removed the duplication.